resources/Microsoft.Windows.Developer/Microsoft.Windows.Developer.psm1

```diff
@@ -59,22 +59,34 @@ class DeveloperMode {
     [DscProperty(NotConfigurable)]
     [bool] $IsEnabled
 
+    hidden [string] $AppModelUnlockRegistryKeyPath = 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\AppModelUnlock\'
+    hidden [string] $DeveloperModePropertyName = 'AllowDevelopmentWithoutDevLicense'
+
     [DeveloperMode] Get() {
-        $this.IsEnabled = IsDeveloperModeEnabled
```

Comment:
I worked to keep IsEnabled and running as admin detection, both functionality that is AFAIK unique to this DSC resource compared to the others. The major contribution to this PR is adding unit tests for DeveloperMode, which required mocks due to the admin detection logic.